collection-scripts/gather_bm_network_logs

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+. gather_service_logs_util
+BASE_COLLECTION_PATH="must-gather"
+NETWORK_LOG_PATH=${OUT:-"${BASE_COLLECTION_PATH}/bm_network_logs"}
+SERVICE_LOG_PATH="${NETWORK_LOG_PATH}"
+
+mkdir -p "${NETWORK_LOG_PATH}"/
+
+function gather_bm_network_logs {
+  POD_INFO=($(oc get pods -A -o=custom-columns=name:.metadata.name,ns:metadata.namespace| grep "${POD_NAME}"))
+  MG_NAMESPACE=${POD_INFO[1]}
+  if [ "${MG_NAMESPACE}" == "ns" ]; then
+    # POD_NAME env variable is passed to must-gather pods since 4.10 version,
+    # if is not defined, command will return the first line of 'get pods' output, which is column name='ns'
+    echo "ERROR: Collection failed, use the latest oc version for it to succeed"
+    exit 1
+  fi
+
+  # Run debug pod on every node first to reduce 'oc wait' time
+  HOST_PODS=()
+  for NODE in ${NODES[@]}; do
+    mkdir -p "${NETWORK_LOG_PATH}/${NODE}"
+    HOST_POD=$(oc debug --to-namespace="${MG_NAMESPACE}" node/"${NODE}" -o jsonpath='{.metadata.name}')
+    oc debug --to-namespace="${MG_NAMESPACE}" node/"${NODE}" -- /bin/bash -c \
+      'trap : TERM INT; sleep infinity & wait' > /dev/null 2>&1 &
+    HOST_PODS+=($HOST_POD)
+  done
+
+  # wait for pod creation
+  sleep 1
+
+  for (( i=0;i<${#NODES[@]};i++ )); do
+    NODE=${NODES[i]}
+    HOST_POD=${HOST_PODS[i]}
+    oc wait -n="${MG_NAMESPACE}" --for=condition=Ready pod/"${HOST_POD}" --timeout=30s
+    if [ $? -ne 0 ]; then
+      echo "ERROR: Timeout waiting for pod to become Ready"
```

Comment:
In addition to logging this error save this to a file in the archive, so when looking at the archive you can quickly see where you did/didn't collect logs from.

Reply:
Empty <node_name> folder should give a hint
I can add erros.txt file like in sosreport to the archive root, or replace empty folder with <node_name.error> or put error file inside a folder. I don't think we have a common way to log errors in must-gather, wdyt?